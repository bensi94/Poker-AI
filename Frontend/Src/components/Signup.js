import React from 'react';
import { Form, Input, Button, Layout } from 'element-react';


class Signup extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            form: {
                username: '',
                password: '',
                confirmPassword: ''
            },
            rules: {
                username: [
                    { required: true, message: 'Please input the username', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the username'));
                        } else {
                            callback();
                        }
                    } }
                ],
                password: [
                    { required: true, message: 'Please input the password', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the password'));
                        } else if (value.length < 5) {
                            callback(new Error('Password must be at least 5 characters'));
                        } else {
                            if (this.state.form.confirmPassword !== '') {
                                this.refs.form.validateField('confirmPassword');
                            }
                            callback();
                        }
                    } }
                ],
                confirmPassword: [
                    { required: true, message: 'Please input the password again', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the password again'));
                        } else if (value !== this.state.form.password) {
                            callback(new Error('The passwords do not match'));
                        } else {
                            callback();
                        }
                    } }
                ]
            }
        };
    }

    handleSubmit(e) {
        e.preventDefault();

        this.refs.form.validate((valid) => {
            if (valid) {
                console.log('isValid');
            } else {
                console.log('invalid');
                return false;
            }
        });
    }

    handleReset(e) {
        e.preventDefault();

        this.refs.form.resetFields();
    }

    onChange(key, value) {
        this.setState({
            form: Object.assign({}, this.state.form, { [key]: value })
        });
    }

    render() {
        return (
            <Layout.Row type="flex" justify="center">
                <Form ref="form"
                    model={this.state.form}
                    rules={this.state.rules}
                    labelWidth="100"
                    className="signup-login-form"
                >
                    <h2 className="signup-login-title">Sign up</h2>
                    <Form.Item label="Username" prop="username">
                        <Input type="text"
                            value={this.state.form.username}
                            onChange={this.onChange.bind(this, 'username')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item label="Pasword" prop="password">
                        <Input type="password"
                            value={this.state.form.password}
                            onChange={this.onChange.bind(this, 'password')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item label="Confirm password" prop="confirmPassword">
                        <Input type="password"
                            value={this.state.form.confirmPassword}
                            onChange={this.onChange.bind(this, 'confirmPassword')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item>
                        <Button type="primary" onClick={this.handleSubmit.bind(this)}>Sign up</Button>
                        <Button onClick={this.handleReset.bind(this)}>Reset</Button>
                    </Form.Item>
                </Form>
            </Layout.Row>
        );
    }
}

export default Signup;
