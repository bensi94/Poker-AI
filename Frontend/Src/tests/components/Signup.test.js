import React from 'react';
import { shallow } from 'enzyme';

import Signup from '../../components/Signup';

describe('Signup Test suite', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Signup />);
    });

    it('Should render Signup(snapshot) correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should change username', () => {
        const usernameVal = 'bensi94';

        // Input field at 0 is the username field, this sets the username
        wrapper.find('Input').at(0).simulate('change', usernameVal);

        expect(wrapper.state('form').username).toBe(usernameVal);
    });

    it('Should change password', () => {
        const passVal = 'testpassword';
        // Input field at 1 is the password field, this sets the password
        wrapper.find('Input').at(1).simulate('change', passVal);

        expect(wrapper.state('form').password).toBe(passVal);
    });

    it('Should call submit on button clicked', () => {
        // This replaces the handleSubmit with a mock function
        // that should be called when the button is clicked
        wrapper.instance().handleSubmit = jest.fn();
        wrapper.instance().forceUpdate();
        wrapper.find('Button').at(0).simulate('click');

        expect(wrapper.instance().handleSubmit).toHaveBeenCalledTimes(1);
    });

    it('Should have a username that is not empty', () => {
        const username = '';
        const callback = jest.fn();

        // Calling the validator with an empty username
        wrapper.instance().state.rules.username[1].validator('', username, callback);

        expect(callback).toHaveBeenCalledTimes(1);
        expect(callback).toHaveBeenCalledWith(new Error('Please input the username'));
    });

    it('Should have password length of at least 5', () => {
        const passVal = 'pw';
        const callback = jest.fn();

        // Calling the validator with a password that is too short
        wrapper.instance().state.rules.password[1].validator('', passVal, callback);

        expect(callback).toHaveBeenCalledTimes(1);
        expect(callback).toHaveBeenCalledWith(new Error('Password must be at least 5 characters'));
    });

    it('Should have matching passwords', () => {
        const passwordVal = 'testpass';
        const confirmPassVal = 'wrongpass';
        const callback = jest.fn();

        // Input field at 1 is the password field, this sets the password
        wrapper.find('Input').at(1).simulate('change', passwordVal);

        // Calling validator in confirmPassword and check if they match
        wrapper.instance().state.rules.confirmPassword[1].validator('', confirmPassVal, callback);

        expect(callback).toHaveBeenCalledTimes(1);
        expect(callback).toHaveBeenCalledWith(new Error('The passwords do not match'));
    });

    it('Should accept valid password', () => {
        const passwordVal = 'testpass';
        const confirmPassVal = 'testpass';
        const callback = jest.fn();

        // Input field at 1 is the password field, this sets the password
        wrapper.find('Input').at(1).simulate('change', passwordVal);

        // Calling the validator with an empty username
        wrapper.instance().state.rules.confirmPassword[1].validator('', confirmPassVal, callback);

        expect(callback).toHaveBeenCalledTimes(1);
        expect(callback).toHaveBeenCalledWith();
    });

    it('Should accept valid username', () => {
        const username = 'bensi94';
        const callback = jest.fn();

        // Calling the validator with an empty username
        wrapper.instance().state.rules.username[1].validator('', username, callback);

        expect(callback).toHaveBeenCalledTimes(1);
        expect(callback).toHaveBeenCalledWith();
    });
});
