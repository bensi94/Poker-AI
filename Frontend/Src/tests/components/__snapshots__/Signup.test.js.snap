// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Test suite Should render Signup(snapshot) correctly 1`] = `
<Row
  align="top"
  justify="center"
  tag="div"
  type="flex"
>
  <Form
    className="signup-login-form"
    labelPosition="right"
    labelSuffix=""
    labelWidth="100"
    model={
      Object {
        "confirmPassword": "",
        "password": "",
        "username": "",
      }
    }
    rules={
      Object {
        "confirmPassword": Array [
          Object {
            "message": "Please input the password again",
            "required": true,
            "trigger": "blur",
          },
          Object {
            "validator": [Function],
          },
        ],
        "password": Array [
          Object {
            "message": "Please input the password",
            "required": true,
            "trigger": "blur",
          },
          Object {
            "validator": [Function],
          },
        ],
        "username": Array [
          Object {
            "message": "Please input the username",
            "required": true,
            "trigger": "blur",
          },
          Object {
            "validator": [Function],
          },
        ],
      }
    }
  >
    <h2
      className="signup-login-title"
    >
      Sign up
    </h2>
    <FormItem
      label="Username"
      prop="username"
    >
      <Input
        autoComplete="off"
        autosize={false}
        onChange={[Function]}
        rows={2}
        trim={false}
        type="text"
        value=""
      />
    </FormItem>
    <FormItem
      label="Pasword"
      prop="password"
    >
      <Input
        autoComplete="off"
        autosize={false}
        onChange={[Function]}
        rows={2}
        trim={false}
        type="password"
        value=""
      />
    </FormItem>
    <FormItem
      label="Confirm password"
      prop="confirmPassword"
    >
      <Input
        autoComplete="off"
        autosize={false}
        onChange={[Function]}
        rows={2}
        trim={false}
        type="password"
        value=""
      />
    </FormItem>
    <FormItem>
      <Button
        disabled={false}
        loading={false}
        nativeType="button"
        onClick={[Function]}
        plain={false}
        type="primary"
      >
        Sign up
      </Button>
      <Button
        disabled={false}
        loading={false}
        nativeType="button"
        onClick={[Function]}
        plain={false}
        type="default"
      >
        Reset
      </Button>
    </FormItem>
  </Form>
</Row>
`;
